### PR TITLE
Fix sheet size for 1-card-per-sheet PDFs and adjust coordinates when using trim

### PR DESCRIPTION
**Pull request by Karneades**
The PR takes the trim value into account when building a PDF from a sprue.

I propose it to `dev` branch and I'm unsure too if this is  only a small issue for one specific use case of mine or if it could help others too...but these changes help building PDFs used for printing for one-card-per-sheet PDFs when using trim. I'm not sure too, whether a trim value for a card size should affect the sheet size...

I got a question from a printing shop if I could provide a 1-card-per-sheet PDF for white cards **without** the bleed (to reduce paper waste).  So the trim function is a great way to trim the card without changing the layout (thanks for your awesome documentation! [Always Have Bleed](https://squib.readthedocs.io/en/latest/bleed.html) and section **Safe and Cut** in [Initial Card Layout](https://squib.readthedocs.io/en/latest/guides/getting-started/part_1_zero_to_game.html) and the DSL method `cut_zone` too).

**When using the built-in sprue for 1 card per sheet** (https://github.com/andymeneely/squib/blob/dev/lib/squib/builtin/sprues/drivethrucards_1up.yml) **the trim value of 1/8 inches** removes the bleed from a card for the PNG image file but **leaves a white border in the PDF (unused space) after trimming.** 

The PR adjusts the coordinates for all cards when there is a trim value and reduce the PDF size according to the trim value for PDFs with 1 card per sheet, so only the the effective card size is used for the sheet size.

### CODE REVIEW

**Comment by andymeneely** [comment]
I like it. I'll do some of my own testing before merging but I like it.

File: lib/squib/graphics/save_sprue.rb [hunk 1]
```diff
@@ -158,11 +160,21 @@ class SaveSpruePDF < SaveSprue
...
         ratio = 72.0 / @deck.dpi
 
-        surface = Cairo::PDFSurface.new(
-          full_filename,
-          @tmpl.sheet_width * ratio,
-          @tmpl.sheet_height * ratio
-        )
+        slots = @tmpl.cards
+        per_sheet = slots.size
+	      if per_sheet == 1
```
Comment (andymeneely) [inline lib/squib/graphics/save_sprue.rb]: Can you change this if-block to this construction?

```ruby
surface = if per_sheet ==1
  Cairo::PDFSurface.new(
  #...
 )
#...
end
```

Comment (Karneades) [reply lib/squib/graphics/save_sprue.rb]: Change it according to your suggestion. And added the change in the save_pdf DSL doc under the sprue option.

**Comment by andymeneely** [comment]
Merged in 1accadfc30bea42d9b08b4d5faa882c269ac9a14

**Pull request closed without merging**